### ISSUE TITLE
he input value type 'ModelStringListFilterInput' is not present when resolving type 'ModelXXXFilterInput'

### ISSUE THREAD

**Issue by jon144** [aws-amplify/amplify-cli#9737]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

14.18.1

### Amplify CLI Version

7.6.19

### What operating system are you using?

Mac

### Did you make any manual changes to the cloud resources managed by Amplify? Please describe the changes made.

No manual changes made

### Amplify Categories

api

### Amplify Commands

push

### Describe the bug

When I push my GraphQL API to the server I get the following error

Resource Name: xxxGraphQLSchema (AWS::AppSync::GraphQLSchema)
Event Type: update
Reason: Schema Creation Status is FAILED with details: Found 1 problem(s) with the schema:
The input value type 'ModelStringListFilterInput' is not present when resolving type 'ModelXXXFilterInput' [@2130:1].
URL: link.... (link took me to a blank screen)

when I remove the column from the schema 'cardOrder' the error goes away.
cardOrder: [String]

Oddly, I have a string array in a different model, and does not cause an error.
columnOrder: [String]

This error happened as I upgraded the Transformer from v1 to v2.  This error did not happen in v1.

### Expected behavior

I expected the graphQL schema to generate code of an array of strings

### Reproduction steps

amplify api push --y

### GraphQL schema(s)

<details>

```graphql
# Put schemas below this line

## this does not produce and error
type KanbanBoard 
  @model(mutations: null, subscriptions: null, queries: null, timestamps: null) 
  @auth(rules: [{allow: groups, groups: ["xxxUsers"]}]) 
{
  id: ID!
  status: String!
  pageId: ID
  page: Page @hasOne(fields: ["pageId"])
  name: String!
  columnOrder: [String]
  columns: [KanbanColumn] @hasMany(indexName: "byBoard", fields: ["id"])
 }

## this schema produces the error
type KanbanColumn 
  @model(mutations: null, subscriptions: null, queries: null, timestamps: null) 
  @auth(rules: [{allow: groups, groups: ["xxxUsers"]}]) 
{
  id: ID!
  kanbanBoardId: ID!  @index(name: "byBoard") 
                      @index(name: "byBoardStatus", sortKeyFields: ["status"])
  status: String!
  title: String
  cardOrder: [String] #when this field is removed the error goes away
  cards: [KanbanCard] @hasMany(indexName: "byColumn", fields: ["id"])
}
```

</details>

### Log output

<details>

```
# Put your logs below this line


```

</details>

**Comment by endoodev1** [comment]
I think I'm having  the same issue with an enum type. Can you check if this is the same problem or if I have to open another issue? 
`# This "input" configures a global authorization rule to enable public access to
# all models in this schema. Learn more about authorization rules here: https://docs.amplify.aws/cli/graphql/authorization-rules
type Location @model 
{
  id: ID!
  name: String! @index(name: "byName")
  users: [LocationUser] @hasMany(indexName: "byLocation", fields: ["id"])
}

type User @model 
{
  id: ID!
  locations: [LocationUser] @hasMany(indexName: "byUser", fields: ["id"])
}

type LocationUser @model(queries: null) 
 {
  id: ID!
  location_id: ID! @index(name: "byLocation", sortKeyFields: ["user_id"])
  location_user_roles: [LocationRole]
  location: Location! @belongsTo(fields: ["location_id"])
}

enum LocationRole {
  ONE
 TWO
}`

Error is: _**The input value type 'ModelLocationRoleListFilterInput' is not present when resolving type 'ModelLocationUserFilterInput'**_

**Comment by endoodev1** [comment]
I think I'm having  the same issue with an enum type. Can you check if this is the same problem or if I have to open another issue?   
```graphql
type Location @model 
{
  id: ID!
  users: [LocationUser] @hasMany(indexName: "byLocation", fields: ["id"])
}

type LocationUser @model(queries: null) 
{
  id: ID!
  location_id: ID! @index(name: "byLocation", sortKeyFields: ["user_id"])
  location_user_roles: [LocationRole]
}

enum LocationRole
{
  ONE
  TWO
}
```

Error is: _**The input value type 'ModelLocationRoleListFilterInput' is not present when resolving type 'ModelLocationUserFilterInput'**_

**Comment by cjihrig** [comment]
@endoodev1 I believe you are encountering the same issue.

**Comment by naedx** [comment]
I am having this same issue using v8.1.0.

When `queries: null` the `input ModelXXXXXInput` type is not generated, yet it is still referenced in the hasMany connection. 

In my case, it is failing with an enum type. Fix #8974 for version 7.6.22 should have resolved this, maybe it regressed?
